app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error, Net::LDAP::Error  => exception
+      exception = Foreman::Exception.new exception.message
+      result[:success] = false
+      result[:error_class] = exception.class.name
+      result[:message] = _(exception.message)
+    rescue => exception
```

Comment:
I'm a little confused as to why there are two rescue blocks with the exact same code? If you plan to handle this in the same way feel free to remove the former one - although I think it's safe to remove the latter `rescue => exception`. As long as we have the proper exceptions on the rescue block it'd be fine.